File: 283.move-zeroes.js
/*
 * @lc app=leetcode id=283 lang=javascript
 *
 * [283] Move Zeroes
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let zero = 0;
  for (const num of nums) {
    if (num !== 0) {
      nums[zero++] = num;
    }
  }
  while (zero < nums.length) {
    nums[zero++] = 0;
  }
};
// @lc code=end

File: 283.move-zeroes.go
/*
 * @lc app=leetcode id=283 lang=golang
 *
 * [283] Move Zeroes
 */

package leetcode

// @lc code=start
func moveZeroes(nums []int) {
	nz := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[i], nums[nz] = nums[nz], nums[i]
			nz++
		}
	}
}

// @lc code=end
